### ISSUE TITLE
Magento 2 When set to developer mode Magento crash 

### ISSUE THREAD

**Issue by testmagento2** [magento/magento2#12030]
After changing magento to dev mode website is crashing and getting php error
And cant understand what is that error about

### Preconditions
magento 2.7
php 70

PHP Fatal error: Uncaught Magento\Framework\Exception\LocalizedException: Invalid Document 
Element 'config', attribute 'xsi:noNamespaceSchemaLocation': The attribute 'xsi:noNamespaceSchemaLocation' is not allowed. 
Line: 2 
in /home/software/public_html/vendor/magento/framework/Config/Reader/Filesystem.php:160 
Stack trace: 
#0 /home/software/public_html/vendor/magento/framework/Config/Reader/Filesystem.php(127): Magento\Framework\Config\Reader\Filesystem->_readFiles(Object(Magento\Framework\Config\FileIterator)) 
#1 /home/software/public_html/vendor/magento/framework/App/ObjectManager/ConfigLoader.php(69): Magento\Framework\Config\Reader\Filesystem->read('global') 
#2 /home/software/public_html/vendor/magento/framework/App/ObjectManager/Environment/Developer.php(77): Magento\Framework\App\ObjectManager\ConfigLoader->load('global') 
#3 /home/software/public_html/vendor/magento/framework/App/ObjectManagerFactory.php(194): Magento\Framework\App\ObjectManager\Environment\Developer->configureObjectManager(Object(Magento\Framework\Inte in /home/software/public_html/vendor/magento/framework/Config/Reader/Filesystem.php on line 160

**Comment by orlangur** [comment]
> PHP Fatal error: Uncaught Magento\Framework\Exception\LocalizedException: Invalid Document
Element 'config', attribute 'xsi:noNamespaceSchemaLocation': The attribute 'xsi:noNamespaceSchemaLocation' is not allowed.

Well, that's expectable, you need to fix invalid (i.e. not corresponding to XSD schema) `config.xml` files.

**Comment by testmagento2** [comment]
How to fix something like that?

**Comment by orlangur** [comment]
Remove such not allowed attribute.

**Comment by gwharton** [comment]
May be related to #10889 which is a similar problem where there is some issue with xml file validation, but only in developer mode. Works fine in production mode.

**Comment by orlangur** [comment]
@gwharton there is no XSD validation in production mode :) Such crash does not indicate that something needs to be changed in Magento but rather in some custom XML configurations.

**Comment by testmagento2** [comment]
Where do I Remove not allowed attribute. I tried to to from the admin I had one that shouldn't be there. I need more help about that ?

Thanks

**Comment by jaromirmuller** [comment]
Is there any general approach how to tackle this problem? 

O have error:
`Fatal error: Uncaught Magento\Framework\Exception\LocalizedException: Invalid Document Element 'config', attribute 'xsi:noNamespaceSchemaLocation': The attribute 'xsi:noNamespaceSchemaLocation' is not allowed. Line: 17 in /var/www/html/vendor/magento/framework/Config/Reader/Filesystem.php on line 163`

And I have now clue where to start finding the issue. Our Magento is 2.2

**Comment by orlangur** [comment]
@jaromirmuller looks like you have something like `<config xsi:noNamespaceSchemaLocation=...` in one of the config XML files.

**Comment by jaromirmuller** [comment]
Is there any "official" way how to debug such issue?

The problem is that the validation validates merged xml object so the information which original file is malformed is not available at the moment of validation.

It's really difficult to find a problem in 5000+ xml files.

Anyway.

After a while I discovered it. But my approach was quite brute force:

I added into vendor/magento/framework/Config/Dom.php logging of validated xml document.

```
public function validate($schemaFileName, &$errors = [])
{
    if ($this->validationState->isValidationRequired()) {

    file_put_contents('dom.xml', $this->dom->saveXML()); // logs into dom.xml
    $errors = $this->validateDomDocument($this->dom, $schemaFileName, $this->errorFormat);
        return !count($errors);
    }
    return true;
}
```

In dom.xml file was a line 17 giving a clue.

One of 3rdparty modules has `<config>` instead of correct `<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">`
